### DISCUSSION TITLE
FR: change default user agent via environment variable

### DISCUSSION THREAD

**Discussion by deanm0000** [encode/httpx#3208] [Ideas]
It'd be a nice convenience if the preferred user-agent could be saved as an environment variable and have that be what is used as the default (it'd still be overridable by changing the header explicitly).

**Comment by lovelydinosaur** [comment]
Interesting yes.

You could use a pattern for that so it's available if you want it tho not default.

```python
UA = os.environ.get(...)
client = httpx.Client(headers=...)
```

Suggest if you want that probably better to be explicit rather than implicit.

**Reply by deanm0000** [reply]
Yeah, of course for developing good code it's better to be explicit but if I'm just in interactive mode then it'd be nice to be able to just `httpx.get` or `post` without preamble.
